File: src/plugin/combat/magic/ancient/blood/BloodBarrageSpellPlugin.java
package plugin.combat.magic.ancient.blood;

import com.rs.game.content.combat.player.style.MagicCombatStyle;
import com.rs.game.entity.actor.Actor;
import com.rs.game.entity.actor.player.Player;
import com.rs.game.plugin.combat.spell.type.CombatSpellPlugin;
import com.rs.utility.constants.MagicConstants.MagicBook;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 8/1/2017
 */
public class BloodBarrageSpellPlugin extends CombatSpellPlugin {
	
	@Override
	public int delay(Player player) {
		return 4;
	}
	
	@Override
	public int animationId() {
		return 1979;
	}
	
	@Override
	public int hitGfx() {
		return 377;
	}
	
	@Override
	public int maxHit(Player player, Actor target) {
		return 290;
	}
	
	@Override
	public void cast(Player source, Actor target, MagicCombatStyle style) {
		style.sendMultiSpell(source, target, this, null, null).forEach(spellContext -> {
			if (spellContext.getHit().getDamage() != 0) {
				source.getPackets().sendGameMessage("You drain some of your opponents' life points.");
				source.heal(spellContext.getHit().getDamage() / 4);
			}
		});
	}
	
	@Override
	public int spellId() {
		return 27;
	}
	
	@Override
	public double exp() {
		return 51;
	}
	
	@Override
	public MagicBook book() {
		return MagicBook.ANCIENTS;
	}
}

File: src/main/java/plugin/interaction/combat/magic/ancient/blood/BloodBarrageSpellPlugin.kt
package plugin.interaction.combat.magic.ancient.blood

import game.content.entity.actor.combat.CombatSwingDetail
import game.content.entity.actor.combat.player.style.MagicCombatStyle
import game.content.plugin.combat.spell.type.CombatSpellPlugin
import game.entity.actor.Actor
import game.entity.actor.player.Player
import utility.constants.MagicConstants.MagicBook
import java.util.function.Consumer

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 8/1/2017
 */
class BloodBarrageSpellPlugin : CombatSpellPlugin {
    override fun delay(player: Player): Int {
        return 4
    }

    override fun animationId(): Int {
        return 1979
    }

    override fun hitGfx(): Int {
        return 377
    }

    override fun maxHit(player: Player, target: Actor): Int {
        return 290
    }

    override fun cast(source: Player, target: Actor, style: MagicCombatStyle) {
        style.sendMultiSpell(source, target, this, null, null).forEach(Consumer { spellContext: CombatSwingDetail ->
            if (spellContext.hit.damage != 0) {
                source.packets.sendMessage("You drain some of your opponents' life points.")
                source.heal(spellContext.hit.damage / 4)
            }
        })
    }

    override fun spellId(): Int {
        return 27
    }

    override fun exp(): Double {
        return 51.0
    }

    override fun book(): MagicBook {
        return MagicBook.ANCIENTS
    }
}